### DISCUSSION TITLE
Unconventional commits missing

### DISCUSSION THREAD

**Discussion by joshka** [orhun/git-cliff#627] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the bug

https://github.com/joshka/tui-scrollview

```
git log --oneline
6766fe2 (HEAD -> main, origin/main, origin/HEAD) docs: update git cliff config
3e71623 docs: ensure that horizontal scrollbar is not shown for example
c1fd2a0 Add horizontal scrollbars and hide scrollbars if there is enough space (#30)
a4217ed (kxxt/main) chore(deps): bump ratatui in the all-dependencies group (#28)
ea12671 ci: group dependabot updates
b3783be chore(deps): bump eyre from 0.6.11 to 0.6.12 (#26)
b437732 (tag: v0.3.4) chore: release (#25)
b194736 fix: scroll_to_bottom scrolls the y offset not the x offset (#24)
1b2eb1e (tag: v0.3.3) chore: release (#19)
```

<details><summary>git cliff result</summary>
<p>

# Changelog

All notable changes to this project will be documented in this file.

### 📚 Documentation

- Ensure that horizontal scrollbar is not shown for example
- Update git cliff config

### ⚙️ Miscellaneous Tasks

- Group dependabot updates

### 🐛 Bug Fixes

- Scroll_to_bottom scrolls the y offset not the x offset (#24)

### ⚙️ Miscellaneous Tasks

- Release (#25)

### 📚 Documentation

- Improve docs and example code (#22)

### ⚙️ Miscellaneous Tasks

- Release (#19)

</p>
</details> 

Note that `c1fd2a0 Add horizontal scrollbars and hide scrollbars if there is enough space (#30)` is missing

filter_unconventional = false in cliff.toml

### Steps To Reproduce

1. `gh repo clone joshka/tui-scrollview`
2. `git cliff` | grep "Add horizontal scrollbars"

### Expected behavior

Changelog should include the unconventional commit

### Software information

- Operating system: macOS
- Rust version: not relevant - git cliff installed via homebrew
- Project version: git-cliff 2.2.1

**Comment by joshka** [comment]
(note: I've force pushed a rewritten commit so that git-cliff behaves nicely, so the repro instructions likely won't work now)

**Comment by orhun** [comment]
(converted to Q&A discussion)

This is the expected behavior, the commit is there if you examine the context via `git cliff -x`:

```json
      {
        "id": "f269bfddac891c9c5a262fabe71d66f7fac16b13",
        "message": "Add horizontal scrollbars and hide scrollbars if there is enough space (#30)\n\n---------\n\nCo-authored-by: Josh McKinney <joshka@users.noreply.github.com>\n",
        "group": null,
        "scope": null,
        "links": [],
        "author": {
          "name": "Levi Zim",
          "email": "rsworktech@outlook.com",
          "timestamp": 1714021585
        },
        "committer": {
          "name": "Orhun Parmaksız",
          "email": "orhunparmaksiz@gmail.com",
          "timestamp": 1714138320
        },
        "conventional": false,
        "merge_commit": false,
        "github": {
          "username": null,
          "pr_title": null,
          "pr_number": null,
          "pr_labels": [],
          "is_first_time": false
        }
      },
```

However in the template:

```
{% for group, commits in commits | group_by(attribute="group") %}
```

And since the group is `null`, the commit is discarded from the changelog. In other words, `git-cliff` does not know where to place this entry.

What you can do is add another parser as follows:

```toml
commit_parsers = [
  { message = ".*", group = "Other" },
]
```

Or update the existing parsers to match that commit:

```toml
commit_parsers = [
  { message = "^Add", group = "Features" },
]
```

**Reply by joshka** [reply]
I did not expect this behavior based on:

1. This is the default configuration of cliff.toml (I ran git cliff --init and then changed the filter-unconventional setting):
2. The docs for that are:
```
# filter out the commits that are not conventional
filter_unconventional = false
```
3. The default configuration also has another setting that would suggest that these shouldn't be filtered:
```
# filter out the commits that are not matched by commit parsers
filter_commits = false
```

These two configuration options combine to suggest that if an unconventional commit is found it's not filtered, and if a commit that doesn't match the parsers is not found it's not filtered. I understand that you're saying that in order for git cliff to include a commit, it has to both be not filtered and also have a section in the parsers that match, but I don't understand what the filter_commits setting does if there's no such section.

The "Other" parser seems like the right approach generally (and perhaps should be added to the default template), but that doesn't resolve the oddness of what the filter_commits setting does.

**Reply by orhun** [reply]
> The "Other" parser seems like the right approach generally (and perhaps should be added to the default template)

Added the "other" parser in 12cb1df561cde39a9a0d0f719156a000f3f4d61b

> These two configuration options combine to suggest that if an unconventional commit is found it's not filtered, and if a commit that doesn't match the parsers is not found it's not filtered. I understand that you're saying that in order for git cliff to include a commit, it has to both be not filtered and also have a section in the parsers that match, but I don't understand what the filter_commits setting does if there's no such section.

Reiterating my point, the commit is not being filtered out by `git-cliff` (you can still see it in the context) but it is being ignored by the template (i.e. `group_by` filter), see:

> "group": null,
> {% for group, commits in commits | group_by(attribute="group") %}

So the unconventional commits are still there but ignored due to this grouping. Maybe we can update this Tera function to allow grouping by null as well. Would be nice to mention it here: https://github.com/Keats/tera/issues/637
